storage/filestore/dir.go

```diff
@@ -116,14 +116,20 @@ func blobPathForFormatVersion(path string, formatVersion storage.FormatVersion)
 	return path + unknownPieceFileSuffix
 }
 
-// blobToTrashPath converts a blob reference to a filepath in transient storage.
-// The files in trash are deleted on an interval (in case the initial deletion didn't work for
-// some reason).
-func (dir *Dir) blobToTrashPath(ref storage.BlobRef) string {
+// blobToTrashPath converts a blob reference to a filepath in transient
+// storage. It will error if dir.garbagedir() does not exist. The files in
+// trash are deleted on an interval (in case the initial deletion didn't work
+// for some reason).
+func (dir *Dir) blobToTrashPath(ref storage.BlobRef) (string, error) {
+	_, err := os.Stat(dir.garbagedir())
```

Comment:
Negatory, sir, I don't think it's appropriate here. `blobToTrashPath` is clearly named as though it is purely functional, with no nondeterministic behavior coming from system state. I'm leery enough of the presence of `dir.garbagedir()`, and that's supposed to be constant. It's worthwhile to keep side effects relegated to the appropriate..... errr, not really monad, but.... let's just say function scope.

As an example consideration, not all callers to `blobToTrashPath()` will care whether the target path already exists.

Sorry to ask for yet another change, but can we move the check to Delete()? Or if that doesn't sit well with you, maybe we can figure out some other way to restructure?

Reply:
Yes, I also didn't like it and was planning to change it tomorrow. I'll move it to `DeleteWithStorageFormat` so we can have the check in both there and `Delete`.

Reply:
@thepaul thank you for a good detailed review, and for not letting me get away with bad code. :-)